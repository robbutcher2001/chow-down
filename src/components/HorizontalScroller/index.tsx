import React, { FunctionComponent, ReactNode } from 'react';

import styled from 'styled-components';

interface HorizontalScrollerProps {
  children: ReactNode
};

const StyledHorizontalScroller = styled.div`
  display: grid;
  grid-template-columns: 100%;
  position: relative;

  &:after {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    pointer-events: none;
    z-index: 10;
    background: ${props => props.theme.isDark ?
      'linear-gradient(to right, rgba(32, 31, 31, 0), rgba(32, 31, 31, 1))' :
      'linear-gradient(to right, rgba(255, 255, 255, 0), rgba(255, 255, 255, 1))'
    };
    width: ${props => props.theme.isDark ?
      '20%' :
      '10%'
    };
  }

  .scroller {
    overflow-x: scroll;
    white-space: nowrap;
    scrollbar-width: none;
    padding: 1rem 0;

    > * {
      margin-right: 0.5rem;
    }

    > :last-child {
      margin-right: 6.25rem;
    }

    &::-webkit-scrollbar {
      display: none;
    }
  }
`

const HorizontalScroller: FunctionComponent<HorizontalScrollerProps> = (props: HorizontalScrollerProps) => {

  return (
    <StyledHorizontalScroller>
      <div className='scroller'>
        {props.children}
      </div>
    </StyledHorizontalScroller>
  );
};

export default HorizontalScroller;