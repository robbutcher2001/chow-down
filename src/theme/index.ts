import { DefaultTheme } from 'styled-components';

import colour from './defs/_colour';

const theme: DefaultTheme = {
  colour
};

export default theme;