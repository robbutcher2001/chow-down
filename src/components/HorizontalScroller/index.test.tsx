import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '../../themes';
import HorizontalScroller from '.';

test('HorizontalScroller basic snapshot render', () => {
  const horizontalScroller = renderer.create(
    <ThemeProvider theme={theme}>
      <HorizontalScroller>
        <div>a child node</div>
      </HorizontalScroller>
    </ThemeProvider>
  );

  expect(horizontalScroller.toJSON()).toMatchSnapshot();
});

test('HorizontalScroller correct tag content assertion', () => {
  const { getByText } = render(
    <ThemeProvider theme={theme}>
      <HorizontalScroller>
        <div>a child node</div>
      </HorizontalScroller>
    </ThemeProvider>
  );

  expect(getByText(/a child node/i)).toBeTruthy();
});