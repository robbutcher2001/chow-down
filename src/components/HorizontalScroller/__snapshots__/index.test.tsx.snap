// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalScroller basic snapshot render 1`] = `
<div
  className="sc-AxjAm cWHdfF"
>
  <div
    className="scroller"
  >
    <div>
      a child node
    </div>
  </div>
</div>
`;
